hack/openapi/main.go

```diff
@@ -0,0 +1,2007 @@
+			ListRef: "#/definitions/kpack.build.v1alpha1.SourceResolverList",
+		}),
+		getCRDPaths(clusterTemplate, CRD{
+			Group:   "build.pivotal.io",
+			Version: "v1alpha1",
+			Kind:    "ClusterBuilder",
+			Plural:  "clusterbuilders",
+			Ref:     "#/definitions/kpack.build.v1alpha1.ClusterBuilder",
+			ListRef: "#/definitions/kpack.build.v1alpha1.ClusterBuilderList",
+		}),
+		getCRDPaths(namespaceTemplate, CRD{
+			Group:   "experimental.pivotal.io",
+			Version: "v1alpha1",
+			Kind:    "CustomBuilder",
+			Plural:  "custombuilders",
+			Ref:     "#/definitions/kpack.experimental.v1alpha1.CustomBuilder",
+			ListRef: "#/definitions/kpack.experimental.v1alpha1.CustomBuilderList",
+		}),
+		getCRDPaths(clusterTemplate, CRD{
+			Group:   "experimental.pivotal.io",
+			Version: "v1alpha1",
+			Kind:    "CustomClusterBuilder",
+			Plural:  "customclusterbuilders",
+			Ref:     "#/definitions/kpack.experimental.v1alpha1.CustomClusterBuilder",
+			ListRef: "#/definitions/kpack.experimental.v1alpha1.CustomClusterBuilderList",
+		}),
+		getCRDPaths(clusterTemplate, CRD{
+			Group:   "experimental.pivotal.io",
+			Version: "v1alpha1",
+			Kind:    "Stack",
+			Plural:  "stacks",
+			Ref:     "#/definitions/kpack.experimental.v1alpha1.Stack",
+			ListRef: "#/definitions/kpack.experimental.v1alpha1.StackList",
+		}),
+		getCRDPaths(clusterTemplate, CRD{
+			Group:   "experimental.pivotal.io",
+			Version: "v1alpha1",
+			Kind:    "Store",
+			Plural:  "stores",
+			Ref:     "#/definitions/kpack.experimental.v1alpha1.Store",
+			ListRef: "#/definitions/kpack.experimental.v1alpha1.StoreList",
+		}),
+	)
+
+	openAPISpec := spec.Swagger{
+		SwaggerProps: spec.SwaggerProps{
+			Swagger: "2.0",
+			Info: &spec.Info{
+				InfoProps: spec.InfoProps{
+					Title:   "kpack",
+					Version: "v0.0.1",
+				},
+			},
+			Paths: &spec.Paths{
+				Paths: paths,
+			},
+			Definitions: combinedOpenAPIDefs,
+		},
+	}
+
+	buf2, err := json.MarshalIndent(openAPISpec, "", "  ")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Print(string(buf2))
+}
+
+func fixName(name string) string {
+	name = strings.ReplaceAll(name, "/", ".")
+	if strings.Contains(name, "github.com.pivotal") {
+		return strings.ReplaceAll(name, "github.com.pivotal.kpack.pkg.apis", "kpack")
+	}
+	if strings.Contains(name, "k8s.io") {
+		return strings.ReplaceAll(name, "k8s.io", "io.k8s")
+	}
+	return name
+}
+
+func getCRDPaths(t *template.Template, crd CRD) map[string]spec.PathItem {
+	buf := &bytes.Buffer{}
+	err := t.Execute(buf, crd)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	paths := map[string]spec.PathItem{}
+	if err := json.Unmarshal(buf.Bytes(), &paths); err != nil {
+		log.Fatal(err)
+	}
+
+	return paths
+}
+
+func combineCRDPaths(paths ...map[string]spec.PathItem) map[string]spec.PathItem {
+	combined := map[string]spec.PathItem{}
+	for _, m := range paths {
+		for k, v := range m {
+			combined[k] = v
+		}
+	}
+	return combined
+}
+
+const k8sOpenAPIDefinitions = `{
```

Comment:
Could we add an information on where this came from?

Reply:
Sure. I'm happy to document this as a comment. In the future, we should probably auto-generate this by querying the kubernetes openapi spec for desired k8s version.